### ISSUE TITLE
browser.evaluate is blocking when browser did not finish creating

### ISSUE THREAD

**Issue by dstainhauser** [maketechnology/chromium.swt#33]
We observe a blocking main thread in the following scenario. Tested with 0.6 and 0.7 on Windows 10, Chromium 64 bit as well on macOS Mojave.

1. A new part is instantiated, that contains a chromium browser widget.
2. In the createControl method, the browser is created.
3. Before the html page is send to the browser.setText(), the window size is being asked from the browser (used to know how to scale).
4. This is done with chromium.evaluate("return window.innerWidth");
5. The call blocks and will never return

On a mac, waiting 200ms before calling browser.evaluate() helps, on Windows not.
So it seems that the evaluate function is blocking when the browser widget is not completely initialized. It happens also sometimes on an initialized widget, after letting the browser completely instantiate.

![image001](https://user-images.githubusercontent.com/1434584/49427232-10e56880-f7a3-11e8-88ce-682baf165e0b.png)

![image001](https://user-images.githubusercontent.com/1434584/49427262-25c1fc00-f7a3-11e8-9753-0894a57e731e.png)

This is the debug log

```
J1:on_address_change:http://text/ true http://text
J1:on_before_browse http://text
J1:on_before_browse:http://localhost/portalV2/loaded http://text
new client: V4(127.0.0.1:59161)
read from socket: 18 24 [1, 0, 0, 0, 0, 0, 0, 0, 1, 3, 1, 0, 0, 0, 0, 0, 49, 0]
st: ReturnSt { kind: Bool, str_value: "1" }
J2:initCef Done empty url
create_browser in 593064
parent 0x90ca8
Calling cef_browser_host_create_browser
J2:on_after_created jnr.ffi.provider.jffi.DirectMemoryIO[address=0x25905580]:null about:blank
J2:GetFocusHandler about:blank
J2:CALLBACK OnSetFocus FOCUS_SOURCE_NAVIGATION about:blank
2018-12-04 08:48:45,632 INFO  RCPBrowser - Created SWT Browser: Chromium
J3:initCef Done empty url
create_browser in 1706754
parent 0x1a0b02
Calling cef_browser_host_create_browser
J3:on_after_created jnr.ffi.provider.jffi.DirectMemoryIO[address=0x25905a80]:null about:blank
J3:GetFocusHandler about:blank
J3:CALLBACK OnSetFocus FOCUS_SOURCE_NAVIGATION about:blank
2018-12-04 08:48:45,686 INFO  RCPBrowser - Created SWT Browser: Chromium
J2:on_after_created handling jnr.ffi.provider.jffi.DirectMemoryIO[address=0x25905580]:null about:blank
J3:on_after_created handling jnr.ffi.provider.jffi.DirectMemoryIO[address=0x25905a80]:null http://text
Evaluating:  window.innerWidth 
```

Many thanks for your help and your good work so far.

**Comment by gzunino** [comment]
@dstainhauser this is going to be fixed in next version likely. Meanwhile you can try calling evaluate in a `ProgressListener.completed` event

**Comment by gzunino** [comment]
@dstainhauser new version 0.8.0 has changes on the way evaluate and browser initialization is handled, could you retest please?

**Comment by dstainhauser** [comment]
Great to hear that this gets fixed with the next version. It occurs from time to time, depending on the timing of events. ProgressListener.completed does not really help, as any Browser.evaluate() function of a JavaScript method that is missing or is not ready will block the entire main thread for ever.

**Comment by dstainhauser** [comment]
The problem is fixed with 0.8.1. Thanks for handling this, you may close the issue.
